### DISCUSSION TITLE
onMount not settable on readable atoms

### DISCUSSION THREAD

**Discussion by matthewhuang97** [pmndrs/jotai#1569] [Q&A]
## Summary

`onMount` is not settable on non-writable atom types. I wonder if this is because `setSelf` is undefined? It's still valuable to use `onMount` in read-only atoms, i.e. if you want to subscribe to the atom's value using `atomStore.sub`. 

## Link to reproduction

https://codesandbox.io/s/youthful-carlos-ezuozj?file=/src/App.tsx

**Comment by dai-shi** [comment]
`onMount` is only attachable to writable atoms.
https://github.com/pmndrs/jotai/blob/785a5f59ee4a291aaded3432cb3747ccd5df2518/src/core/atom.ts#L64-L71

It's by design. If you could overwrite a read-only atoms, it means it would break the atom dataflow graph.

> if you want to subscribe to the atom's value using `atomStore.sub`.

I think something is possible with using multiple atoms.
I'm not sure if I understand the specific use case. (codesandbox doesn't have code?)
